### PR TITLE
Add update service

### PR DESCRIPTION
**Pull request by davis999**

### CODE REVIEW

File: src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java [hunk 1]
```diff
@@ -76,4 +92,47 @@ public ProductView createProduct(ProductDraft productDraft) {
...
+   * Update product entity.
+   *
+   * @param entity  the product entity
+   * @param actions the actions
+   * @return product
+   */
+  @Transactional
+  private Product updateProductEntity(Product entity, List<UpdateAction> actions) {
+    LOG.debug("Enter. ProductId: {}, actions: {}.", entity.getId(), actions);
+    actions.parallelStream().forEach(action -> {
```
Comment (kaka89) [inline src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java]: Use stream

File: src/main/java/io/reactivesw/product/domain/service/ProductService.java [hunk 2]
```diff
@@ -130,6 +130,22 @@ public Product getProductById(String id) {
...
+   * @param entity the entity
+   * @return the product
+   */
+  public Product save(Product entity) {
+    LOG.debug("Enter. ProductId: {}.", entity.getId());
+    LOG.trace("Product: {}.", entity);
+
+    Product savedEntity = productRepository.save(entity);
+
+    LOG.debug("Exit");
```
Comment (kaka89) [inline src/main/java/io/reactivesw/product/domain/service/ProductService.java]: .

File: src/main/java/io/reactivesw/product/infrastructure/update/UpdateAction.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+/**
+ * Configurations for common update actions that will be used in more thant one service
+ * and this action also extends other action configure in each service.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "action")
+@JsonSubTypes({})
```
Comment (kaka89) [inline src/main/java/io/reactivesw/product/infrastructure/update/UpdateAction.java:11]: empty?

File: src/main/java/io/reactivesw/product/infrastructure/validator/VersionValidator.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  /**
+   * Validate entity version.
+   *
+   * @param entity  the entity
+   * @param version the version
+   * @throws ConflictException if version not equal
+   */
+  public static void validate(Product entity, Integer version) {
+    if (!Objects.equals(entity.getVersion(), version)) {
```
Comment (kaka89) [inline src/main/java/io/reactivesw/product/infrastructure/validator/VersionValidator.java]: don't use `Objects.equals` here

File: src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java (hunk 1 above)
Comment (davis999) [reply src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java]: done

File: src/main/java/io/reactivesw/product/domain/service/ProductService.java (hunk 2 above)
Comment (davis999) [reply src/main/java/io/reactivesw/product/domain/service/ProductService.java]: done

File: src/main/java/io/reactivesw/product/infrastructure/update/UpdateAction.java (hunk 3 above)
Comment (davis999) [reply src/main/java/io/reactivesw/product/infrastructure/update/UpdateAction.java:11]: yep, do not add in this pr.

File: src/main/java/io/reactivesw/product/infrastructure/validator/VersionValidator.java (hunk 4 above)
Comment (davis999) [reply src/main/java/io/reactivesw/product/infrastructure/validator/VersionValidator.java]: done

**Review by kaka89 — APPROVED** [review]

**Pull request merged**